src/main/java/com/paneedah/mwc/items/guns/M79Factory.java

```diff
@@ -1,612 +1,173 @@
 package com.paneedah.mwc.items.guns;
 
 import com.paneedah.mwc.MWC;
-import com.paneedah.mwc.models.weapons.M79;
+import com.paneedah.mwc.models.weapons.*;
 import com.paneedah.mwc.proxies.CommonProxy;
-import com.paneedah.mwc.weapons.Attachments;
 import com.paneedah.mwc.weapons.AuxiliaryAttachments;
 import com.paneedah.mwc.weapons.Bullets;
-import com.paneedah.weaponlib.RenderableState;
-import com.paneedah.weaponlib.Weapon;
-import com.paneedah.weaponlib.WeaponRenderer;
-import com.paneedah.weaponlib.animation.Transition;
+import com.paneedah.weaponlib.*;
+import com.paneedah.mwc.rendering.Transform;
+import com.paneedah.weaponlib.compatibility.RecoilParam;
 import net.minecraft.item.Item;
 import org.lwjgl.opengl.GL11;
 
-import java.util.Arrays;
-
 public class M79Factory implements GunFactory {
 
     public Item createGun(CommonProxy commonProxy) {
         return new Weapon.Builder()
+                .withName("m79")
+                .withAmmoCapacity(1)
+                .withFireRate(0.1f)
```

Comment:
No balance changes until 0.2.
```suggestion
                .withFireRate(0.7f)
```